go/prometheusexporter/collector/collector.go

```diff
@@ -171,6 +212,7 @@ func (c *Collector) observeMetrics(m *metric) {
 func (c *Collector) observer() {
 	for !util.Interrupted() && atomic.LoadUint32(&c.stopped) == 0 {
 		m := <-c.metricsChannel
+		c.mutex.Lock()
```

Comment:
Shouldn't there be a `defer c.mutex.Unlock()` after this line? Searching through the entire file the calls between mutex.Lock and mutex.Unlock seem unbalanced (off by one).

Reply:
This is a left over from refactoring the code mid change. Thx for spotting!